### PR TITLE
[SVS-2] Add encrypted affidavit to castRequestItem

### PR DESCRIPTION
**Pull request by w0st**

### CODE REVIEW

File: lib/av_client.ts [hunk 1]
```diff
@@ -393,11 +394,24 @@ export class AVClient implements IAVClient {
           content: {}
       };
 
+      if (affidavit && this.electionConfig) {
+        try {
+          const encryptedAffidavit = JSON.parse(dhEncrypt(this.electionConfig.castRequestItemAttachmentEncryptionKey, affidavit).toString())
+          console.log(encryptedAffidavit)
+          castRequestItem.content['attachment'] = encryptedAffidavit.ciphertext
+          castRequestItem['attachment'] = `data:text/plain;base64,${encryptedAffidavit.ciphertext}`
```
Comment (w0st) [inline lib/av_client.ts:402]:
```suggestion
          const encryptedAffidavit = dhEncrypt(this.electionConfig.castRequestItemAttachmentEncryptionKey, affidavit).toString()
          console.log(encryptedAffidavit)
          castRequestItem.content['attachment'] = sha256(encryptedAffidavit)
          castRequestItem['attachment'] = `data:text/plain;base64,${encryptedAffidavit}`
```

`dhEncrypt#toString()` returns stringified (serialized) object, i suppose it should be easy to deserialize it on BE by using `Crypto::AES::DHPackage.from_string(encryptedAffidavit)`

@av-emil Could you confirm, please?

**Comment by w0st** [comment]
It works :tada: 
Thanks!

File: lib/av_client.ts (hunk 1 above)
Comment (av-emil) [reply lib/av_client.ts]: This snippet is outdated. Do you still need my input, @w0st?

Comment (w0st) [reply lib/av_client.ts]: Thanks, I'm fine now. The current solution works correctly with [this approach to decrypt (How to decrypt affidavits?
 @ Slack)](https://monterail.slack.com/archives/C0421MSKUKS/p1669216669333509?thread_ts=1669216475.509959&cid=C0421MSKUKS).

Comment (av-emil) [reply lib/av_client.ts]: @w0st, that slack link is on the monterail domain. I don't believe I have access there. 😅

File: lib/av_client/types.ts [hunk 2]
```diff
@@ -271,6 +271,8 @@ export interface ElectionConfig {
   // appended data:
   affidavit: AffidavitConfig;
 
+  castRequestItemAttachmentEncryptionKey: string;
```
Comment (w0st) [inline lib/av_client/types.ts:274]: The current implementation of `js-client` points to `http://dbb:3003/:election_slug/configuration` which doesn't return `castRequestItemAttachmentEncryptionKey` in response.

The `castRequestItemAttachmentEncryptionKey`  is returned only by `http://dbb:3003/:election_slug/configuration/latest_config` (see: https://github.com/aion-dk/dbb/pull/146)

@av-emil What would you like to do with it?

File: lib/av_client.ts (hunk 1 above)
Comment (w0st) [reply lib/av_client.ts]: I wanted to show you this message image

File: lib/av_client/types.ts (hunk 2 above)
Comment (av-emil) [reply lib/av_client/types.ts:274]: @av-lukas and/or @av-mads, what's the intented difference between `dbb:3003/:election_slug/configuration` and `dbb:3003/:election_slug/configuration/latest_config`?

File: lib/av_client.ts [hunk 3]
```diff
@@ -325,6 +327,10 @@ export class AVClient implements IAVClient {
       contestEnvelopes,
     } = constructContestEnvelopes(state, ballotSelection);
 
+    console.log('%%%%%')
+    console.log(contestEnvelopes)
+    console.log('%%%%%')
```
Comment (av-emil) [inline lib/av_client.ts:332]: Remember to remove this before merging.

File: lib/av_client.ts [hunk 4]
```diff
@@ -343,7 +349,10 @@ export class AVClient implements IAVClient {
       this.privateKey(),
       this.getDbbPublicKey()
     );
-
+    console.log('$$$$$$$')
+    console.log(serverEnvelopes)
+    console.log(boardCommitment)
+    console.log('$$$$$$$')
```
Comment (av-emil) [inline lib/av_client.ts:355]: Remember to remove this before merging.

**Pull request merged**